example_web/index.html

```diff
@@ -0,0 +1,94 @@
+<html>
+  <head>
+    <meta charset="utf-8"/>
+    <title>omggif</title>
+  </head>
+
+  <body>
+    <h1>omggif</h1>
+
+    <input type="file" id="file" accept="image/gif">
+
+    <div>
+      <canvas id="player"></canvas>
+    </div>
+
+    <script type="text/javascript" src="../omggif.js"></script>
+    <script type="text/javascript">
+      var reader, context, frameNumber, previousFrameInfo, loops, timeout;
+
+      function parseGif(array) {
+        reader = new GifReader(array);
+
+        frameNumber = 0;
+        loops = reader.loopCount();
+        previousFrameInfo = null;
+
+        clearTimeout(timeout);
+        var canvas = document.querySelector('#player');
+        canvas.width = reader.width;
+        canvas.height = reader.height;
+        context = canvas.getContext('2d');
+
+        draw();
+      }
+
+      function draw() {
+        if (!reader) { return; }
+
+        var frameInfo = reader.frameInfo(frameNumber);
+
+        if (frameNumber === 0) {
+          // always clear canvas on first frame
+          context.clearRect(0, 0, reader.width, reader.height);
+        } else if (previousFrameInfo && previousFrameInfo.disposal === 2) {
+          // disposal was "restore to background" which is essentially
+          // "restore to transparent"
+          context.clearRect(previousFrameInfo.x,
```

Comment:
It's been a long time, I would have to refresh myself on all of the disposal stuff again, but I am pretty sure to do it properly you need to be able to fill with a background colour and stuff like that.  I know that my plask example didn't do this either, maybe this is a chance for me to refresh myself on the details again...

Reply:
Yes, there were a number of disposal methods in my original code that weren't quite right but should be fixed now. For background disposal, in practice all browsers dispose to transparent so I did that here as well.